doc/src/dev_intro.rst

```diff
@@ -0,0 +1,138 @@
+MIGraphX Fundamentals
+======================
+
+MIGraphX provides an optimized execution engine for deep learning neural networks.
+In this introduction for developers of MIGraphX, we will cover some simple operations in the MIGraphX framework.
+
+
+Location of the Examples
+-------------------------
+
+The ``ref_dev_examples.cpp`` can be found in the test directory (``/test``).
+The executable file ``test_ref_dev_examples`` based on this file will be created in the ``bin/`` of the build directory after running ``make -j$(nproc) check``.
```

Comment:
I don't think we should use make check if the user only wants to run the dev examples.